### ISSUE TITLE
dYdx Perpetual - Error submitting Sell/Buy LIMIT Order. 

### ISSUE THREAD

**Issue by sidmontanez** [CoinAlpha/hummingbot#3450]
**Describe the bug**
-User everSin#8734 reported the issue about his Bot that fails to open an order when running on a `perpetual_market_making` strategy.  

**Steps To Reproduce**
-Create a `perpetual_market_making` strategy
-Choose `dYdX Perpetual` as your derivative exchange
-Choose `1INCH-USD` as your trading pair. 
-`start` the bot and upon creating an order, It would throw out "Error submitting SELL/BUY LIMIT order" 

Here are the steps to reproduce the issue (see attachments in the section below):
1. Set up your configuration file with these parameters:

![](https://cdn.discordapp.com/attachments/772104796789932033/847168774573785138/image0.png)

![](https://cdn.discordapp.com/attachments/772104796789932033/847168877867565076/image0.png)

**Screenshots**
// If applicable, add screenshots to help explain your problem.
![image](https://user-images.githubusercontent.com/78310937/119717159-87f1a880-be98-11eb-81f1-2194b68bac13.png)

**Release version**
Version 0.39.0

**Attachments**
[logs_conf_perpetual_market_making_1Inch_1 (1).log](https://github.com/CoinAlpha/hummingbot/files/6549118/logs_conf_perpetual_market_making_1Inch_1.1.log)

**Comment by sidmontanez** [comment]
Upon further testing by the QA @dendenski .  Only 3 Trading Pairs only works while running the strategy in dYdX. 
```
BTC-USD     OK
CRV-USD     FAIL
DOGE-USD    FAIL
SUSHI-USD   FAIL
YFI-USD     FAIL
UNI-USD     FAIL
AVAX-USD    FAIL
ETH-USD     OK
SNX-USD     FAIL
1INCH-USD   FAIL
LINK-USD    OK
AAVE-USD    FAIL
SOL-USD     FAIL
```

**Comment by RC-13** [comment]
Adding traceback error:
```python
2021-05-27 06:09:22,597 - 1591 - hummingbot.connector.derivative.dydx_perpetual.dydx_perpetual_derivative - WARNING - Error submitting SELL LIMIT order to dydx for 10 1INCH-USD at 3.326.
2021-05-27 06:09:22,597 - 1591 - hummingbot.connector.derivative.dydx_perpetual.dydx_perpetual_derivative - INFO - '1INCH-USD'
Traceback (most recent call last):
  File "/Users/clint_g/Documents/Github-Repo/hb-general/hummingbot/connector/derivative/dydx_perpetual/dydx_perpetual_derivative.py", line 357, in execute_order
    creation_response = await self.place_order(client_order_id, trading_pair, amount,
  File "/Users/clint_g/Documents/Github-Repo/hb-general/hummingbot/connector/derivative/dydx_perpetual/dydx_perpetual_derivative.py", line 305, in place_order
    return await self.dydx_client.place_order(
  File "/Users/clint_g/Documents/Github-Repo/hb-general/hummingbot/connector/derivative/dydx_perpetual/dydx_perpetual_client_wrapper.py", line 60, in place_order
    return await f
  File "/Users/clint_g/opt/anaconda3/envs/hummingbot/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/clint_g/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/dydx3/modules/private.py", line 485, in create_order
    order_to_sign = SignableOrder(
  File "/Users/clint_g/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/dydx3/starkex/order.py", line 58, in __init__
    synthetic_asset = SYNTHETIC_ASSET_MAP[market]
KeyError: '1INCH-USD'
```

**Comment by mifeng** [comment]
I see something similar for `UNI-USD`:
```
2021-07-14 15:09:48,531 - 8982 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing BUY order for 1.0 UNI at dydx_perpetual at 17.899 price to OPEN position. [clock=2021-07-14 22:09:48+00:00]
2021-07-14 15:09:48,817 - 8982 - hummingbot.connector.derivative.dydx_perpetual.dydx_perpetual_derivative - WARNING - Error submitting BUY LIMIT order to dydx for 1.0 UNI-USD at 17.899.
2021-07-14 15:09:48,818 - 8982 - hummingbot.connector.derivative.dydx_perpetual.dydx_perpetual_derivative - INFO - 'UNI-USD'
Traceback (most recent call last):
  File "/Users/feng/Code/hummingbot/hummingbot/connector/derivative/dydx_perpetual/dydx_perpetual_derivative.py", line 355, in execute_order
    creation_response = await self.place_order(client_order_id, trading_pair, amount,
  File "/Users/feng/Code/hummingbot/hummingbot/connector/derivative/dydx_perpetual/dydx_perpetual_derivative.py", line 303, in place_order
    return await self.dydx_client.place_order(
  File "/Users/feng/Code/hummingbot/hummingbot/connector/derivative/dydx_perpetual/dydx_perpetual_client_wrapper.py", line 60, in place_order
    return await f
  File "/Users/feng/anaconda3/envs/hummingbot/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/feng/anaconda3/envs/hummingbot/lib/python3.8/site-packages/dydx3/modules/private.py", line 485, in create_order
    order_to_sign = SignableOrder(
  File "/Users/feng/anaconda3/envs/hummingbot/lib/python3.8/site-packages/dydx3/starkex/order.py", line 58, in __init__
    synthetic_asset = SYNTHETIC_ASSET_MAP[market]
KeyError: 'UNI-USD'
2021-07-14 15:09:48,818 - 8982 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Retrying failed order on derivative exchange.
2021-07-14 15:09:48,829 - 8982 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626300588000, "type": "OrderType.LIMIT", "trading_pair": "UNI-USD", "amount": "1.0", "price": "17.899", "order_id": "buy-UNI-USD-1626300588531483", "exchange_order_id": null, "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "dydx_perpetual"}
```

**Comment by dennisocana** [comment]
Changing priority status from P2 to P1

**Comment by dennisocana** [comment]
Fixed in version 0.42 release.
